packages/graph/hash_graph/lib/graph/src/store/postgres/knowledge/link/read.rs

```diff
@@ -140,34 +152,84 @@ async fn by_link_type_by_source_entity_id(
-
-    async fn read<'query>(
-        &self,
-        query: &Self::Query<'query>,
-    ) -> Result<Vec<OutgoingLinks>, QueryError> {
-        match (query.link_type_query(), query.source_entity_id()) {
-            (None, Some(source_entity_id)) => {
-                by_source_entity_id(self.as_client(), source_entity_id).await
-            }
-            (Some(link_type_query), Some(source_entity_id)) => {
-                match (link_type_query.uri(), link_type_query.version()) {
-                    (Some(link_type_uri), Some(OntologyVersion::Exact(link_type_version))) => {
-                        by_link_type_by_source_entity_id(
-                            self.as_client(),
-                            link_type_uri,
-                            link_type_version,
-                            source_entity_id,
-                        )
-                        .await
+impl<C> Resolve<C> for LinkRecord
+where
+    C: PostgresContext + Sync,
+{
+    async fn resolve(&self, path: &[PathSegment], context: &C) -> Result<Literal, ResolveError> {
+        match path {
+            [] => todo!("{}", UNIMPLEMENTED_LITERAL_OBJECT),
+            [head_path_segment, tail_path_segments @ ..] => {
+                let literal = match head_path_segment.identifier.as_str() {
+                    "type" => {
+                        return context
+                            .read_versioned_ontology_type::<LinkType>(&self.type_uri)
+                            .await
+                            .change_context(ResolveError::StoreReadError)?
+                            .resolve(tail_path_segments, context)
+                            .await;
+                    }
+                    "source" => {
+                        return context
+                            .read_latest_entity_by_id(self.source_entity_id)
+                            .await
+                            .change_context(ResolveError::StoreReadError)?
+                            .resolve(tail_path_segments, context)
+                            .await;
                     }
-                    _ => todo!(),
+                    "target" => {
+                        return context
+                            .read_latest_entity_by_id(self.target_entity_id)
+                            .await
+                            .change_context(ResolveError::StoreReadError)?
+                            .resolve(tail_path_segments, context)
+                            .await;
+                    }
+                    "active" => Literal::Bool(self.is_active),
+                    _ => Literal::Null,
+                };
+
+                if tail_path_segments.is_empty() {
+                    Ok(literal)
+                } else {
+                    literal.resolve(tail_path_segments, context).await
                 }
             }
-            _ => todo!(),
         }
     }
 }
+
+// TODO: we should probably support taking PersistedEntityIdentifier here as well as an EntityId
```

Comment:
Is this `TODO` still relevant here? This would be handled by underlying the resolvers I assume

Reply:
I think you are right. In the case we need it anyway we can still add it later on but I currently don't see a reason why we would need it here.
```suggestion
```